autoload/codeium/server.vim

```diff
@@ -122,42 +122,50 @@ function! s:SendHeartbeat(timer) abort
 endfunction
 
 function! codeium#server#Start(...) abort
-  silent let os = substitute(system('uname'), '\n', '', '')
-  silent let arch = substitute(system('uname -m'), '\n', '', '')
-  let is_arm = stridx(arch, 'arm') == 0 || stridx(arch, 'aarch64') == 0
-
-  if os ==# 'Linux' && is_arm
-    let bin_suffix = 'linux_arm'
-  elseif os ==# 'Linux'
-    let bin_suffix = 'linux_x64'
-  elseif os ==# 'Darwin' && is_arm
-    let bin_suffix = 'macos_arm'
-  elseif os ==# 'Darwin'
-    let bin_suffix = 'macos_x64'
-  else
-    let bin_suffix = 'windows_x64.exe'
-  endif
+  let user_defined_codeium_bin = get(g:, 'codeium_ls_binary', '')
 
-  let sha = get(codeium#command#LoadConfig(codeium#command#XdgConfigDir()), 'sha', s:language_server_sha)
-  let bin_dir = codeium#command#HomeDir() . '/bin/' . sha
-  let s:bin = bin_dir . '/language_server_' . bin_suffix
-  call mkdir(bin_dir, 'p')
+  if user_defined_codeium_bin != '' && filereadable(user_defined_codeium_bin)
+    let s:bin = user_defined_codeium_bin
+    call s:ActuallyStart()
```

Comment:
Simpler to call return after this than add the else and reindent the rest of the code.

Reply:
Indeed ! Done